### ISSUE TITLE
Refactor RM code

### ISSUE THREAD

**Issue by sanagno** [LAION-AI/Open-Assistant#2049]
RM codebase is outdated. We should consider refactoring it.

I propose moving everything together with the SFT/RL training to reuse modules/utils.

**Comment by CloseChoice** [comment]
So I would suggest the following refactorings:
  - the get_train_loader method is exactly the same is exactly the same for SFTTrainer and RMTrainer. So we could refactor this into a TrainerMixin.
  - setting the configs works exactly the same from the parameters. This could be refactored to a utils function.
  - building the sampler/ initializing trainingargs from config is also identical -> refactor to utils

If that sounds find for you @sanagno I would be willing to do the refactoring.

**Comment by sanagno** [comment]
Hey @CloseChoice, the reward modelling code is already refactored under model_training if this is what you mean.

The plan is to copy any additional files needed from `reward/instructor`, then everything under that can just be deleted.

**Comment by CloseChoice** [comment]
I see. What about the things I pointed out. Are these also things you consider for refactoring, e.g. the duplicated `get_train_loader` method for SFTTrainer and RMTrainer?

I'd like to the refactoring from `reward/instructor` and move the files to `model_training`
